src/lib/svg_text_utils.js

```diff
@@ -40,18 +40,23 @@ exports.convertToTspans = function(_context, gd, _callback) {
     svgClass += '-math';
     parent.selectAll('svg.' + svgClass).remove();
     parent.selectAll('g.' + svgClass + '-group').remove();
-    _context.style({visibility: null});
+    _context.style({visibility: null})
+        .attr({
+            // some callers use data-unformatted *from the <text> element* in 'cancel'
+            // so we need it here even if we're going to turn it into math
+            // these two (plus style and text-anchor attributes) form the key we're
+            // going to use for Drawing.bBox
+            'data-unformatted': str,
+            'data-math': ''
+        });
 
     function showText() {
         if(!parent.empty()) {
             svgClass = _context.attr('class') + '-math';
             parent.select('svg.' + svgClass).remove();
         }
         _context.text('')
-            .style({
-                visibility: 'inherit',
```

Comment:
Ah, I didn't notice the original issue #984 that led to this change... assumed it was a remnant of a much older time. My goal was to *remove* inline styles whenever possible, rather than setting an explicit reset value, but anyway I [moved](https://github.com/plotly/plotly.js/pull/1792/commits/61dd3b181a9611a24651ce6c88f8f075438e899e) from `visibility:hidden` to `display:none` because only the latter takes the item out of the bounding box, and there are still a couple of places (most notably titles/index:146 & 171 that I'm still working on) we get the bounding box of the combined text+MathJax element. I'll see if I can come up with a test of #984 to include with this PR.